src/database/datatools/designday.py

```
@@ -0,0 +1,234 @@
+                                  sky_clearness: Optional[float] = None,
+                                  maximum_number_warmup_days: Optional[int] = None,
+                                  begin_environment_reset_mode: Optional[str] = None) -> None:
+    
+    conn = sqlite3.connect(db_path)
+    conn.row_factory = sqlite3.Row
+    cursor = conn.cursor()
+    cursor.execute("SELECT * FROM sizingperiod_designday WHERE id = ?", (designday_id,))
+    row = cursor.fetchone()
+    if row is None:
+        conn.close()
+        raise ValueError(f"SizingPeriod:DesignDay with id {designday_id} does not exist.")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)